duoapi.go

```diff
@@ -197,17 +218,27 @@ func (duoapi *DuoApi) Call(method string,
 		Path:     uri,
 		RawQuery: params.Encode(),
 	}
-	request, err := http.NewRequest(method, url.String(), nil)
-	if err != nil {
-		return nil, nil, err
-	}
-	resp, err := client.Do(request)
-	var body []byte
-	if err == nil {
-		body, err = ioutil.ReadAll(resp.Body)
-		resp.Body.Close()
+
+	backoffMs := initialBackoffMS
+	for {
+		request, err := http.NewRequest(method, url.String(), nil)
+		if err != nil {
+			return nil, nil, err
+		}
+		resp, err := client.Do(request)
```

Comment:
I personally think it would help for code clarity to put the err check here as opposed to inside the if below.

Reply:
There is no reason or desire to read the body in the case of a 429 status code, so moving it outside of the if would create extra unnecessary work

Reply:
Are you sure you always get a response back even on non-nil err?

https://golang.org/pkg/net/http/#Client.Do
```
On error, any Response can be ignored. A non-nil Response with a non-nil error only occurs when CheckRedirect fails, and even then the returned Response.Body is already closed.
```

Reply:
👍 great catch

Reply:
Let me know if this update addresses that concern.  I added a test in as well for that case

Reply:
That's awesome. Thanks!